opm/autodiff/NewtonIterationBlackoilInterleaved.cpp

```diff
@@ -153,8 +153,10 @@ namespace Opm
             const ParallelISTLInformation& info =
                 boost::any_cast<const ParallelISTLInformation&>( parallelInformation_);
             Comm istlComm(info.communicator());
+            // As we use a dune-istl with block size np the number of components
+            // per parallel is only one.
```

Comment:
Let's give it another try:

The parallelization of dune-istl exspects one pair of global (unique
over all processes) and local (0, ..., number of indices on process)
index per block. For the interleaved version this block represents all
phases. For the non-interleaved one it represents just one phase.

The indexset passed to copy values contains one index pair per
cell. This is sufficient for the interleaved solver with one block per
cell.

But for the other one we need to communicates number of phases (which
I call number_of_components here) values which is not possible for
that index set. Therefore we create a new one that has
number_of_components times the entries and use that one for the
communication.
-- 
Join us at the DUNE User Meeting Sept. 28-29, 2015, Heidelberg, Germany
http://users.dune-project.org/projects/dune-user-meeting-2015/wiki